### DISCUSSION TITLE
Step 6 script `Build ipa for distribution` exited with status code 65

### DISCUSSION THREAD

**Discussion by ariefrhmn** [codemagic-ci-cd/codemagic-docs#2801] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

build ID: 672067af8684aa574154f760

[DiriClinic 2.log](https://github.com/user-attachments/files/17551760/DiriClinic.2.log)

I just not understand what caused error. anyone help me, please. thanks!

### Expected results

build success

### Actual results

Failed to archive /Users/builder/clone/ios/DiriClinic.xcworkspace

Build failed :|
Step 6 script `Build ipa for distribution` exited with status code 65

### Build id (optional)

672067af8684aa574154f760

**Comment by ariefrhmn** [comment]
hello, thread above is solved by adding `--archive-flags="-destination 'generic/platform=iOS'"` in the script.

but, after I add new Scheme config for my project, i've got same error.
here is the build Id:
`672db2d8d1a257bd2b803fd0`

i'm attached the log
[DiriClinic 4.log](https://github.com/user-attachments/files/17674501/DiriClinic.4.log)

I just not understand what caused error. anyone help me, please. thanks!

**Reply by ariefrhmn** [reply]
now i add `--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"` to the script.
now error exited with status code 9.

build Id 
`672db5f8d1a257bd2b804b8d`

[DiriClinic 6.log](https://github.com/user-attachments/files/17674808/DiriClinic.6.log)

more reference error:
```shell
[07:24:19] WARNING > Executing "xcode-project build-ipa" failed unexpectedly. Detailed logs are available at "/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/codemagic-08-11-24.log". 
[07:24:19] ERROR > Exception traceback:
Traceback (most recent call last):
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/models/code_sign_entitlements.py", line 70, in from_app
    process.raise_for_returncode()
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/cli/cli_process.py", line 121, in raise_for_returncode
    raise subprocess.CalledProcessError(self.returncode, self._command_args, stdout, stderr)
subprocess.CalledProcessError: Command '('codesign', '--display', '--entitlements', '-', '--xml', 'build/ios/xcarchive/DiriClinic_6a8pufg3.xcarchive/Products/Applications/DiriClinicDevelopment.app')' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/cli/cli_app.py", line 243, in invoke_cli
    CliApp._running_app._invoke_action(args)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/cli/cli_app.py", line 184, in _invoke_action
    return cli_action(**action_args)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/cli/action.py", line 83, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/tools/xcode_project.py", line 284, in build_ipa
    self._update_export_options(xcarchive, export_options_plist, export_options)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/tools/xcode_project.py", line 799, in _update_export_options
    archive_entitlements = CodeSignEntitlements.from_xcarchive(xcarchive)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/models/code_sign_entitlements.py", line 96, in from_xcarchive
    return cls.from_app(app_path)
  File "/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/models/code_sign_entitlements.py", line 75, in from_app
    raise IOError(f"Failed to obtain entitlements from {app_path}, {cls._str(cpe.stderr)}")
OSError: Failed to obtain entitlements from build/ios/xcarchive/DiriClinic_6a8pufg3.xcarchive/Products/Applications/DiriClinicDevelopment.app, build/ios/xcarchive/DiriClinic_6a8pufg3.xcarchive/Products/Applications/DiriClinicDevelopment.app: code object is not signed at all

/Applications/Xcode-15.2.app/Contents/Developer


Build failed :|
Step 6 script `Build ipa for distribution` exited with status code 9
``` 
thank you

**Reply by dtrdic** [reply]
Hi @ariefrhmn if you download the produced zip file under the artifact sections you will see some additional logs, and inside you will see this errors
```
DiriClinic.xcodeproj: error: Provisioning profile "NotificationServices Dev" doesn't match the entitlements file's value for the com.apple.security.application-groups entitlement.
DiriClinic.xcodeproj: error: Provisioning profile "Diri Care Dev" doesn't match the entitlements file's value for the com.apple.security.application-groups entitlement.
```
So the error indicates that the provisioning profiles you’re using do not match the required app group entitlement defined in your project.
Doublechekc that the entitlements file (.entitlements) in your Xcode project contains the correct app group identifier(s)
<img width="446" alt="image" src="https://github.com/user-attachments/assets/926bef15-04f1-4a0b-b468-7378b51c78e2">

And make sure this identifiers match the one you have set in the [Apple developer portal](https://developer.apple.com/account/resources/identifiers/list)
![image](https://github.com/user-attachments/assets/1ce8f799-c7ce-497f-91ce-4a7ea156a039)

After you double-checked, the safest way is to re-download your provisioning profiles from Apple developer portal and upload it to Codemagic

**Reply by ariefrhmn** [reply]
i've checked many times, and already do the re-download provisioning profiles from Apple developer portal and upload it to Codemagic. still get the same **error**, even fetch from Apple Developer Portal.
I'm pretty sure the entitlements / appgroup thingy is correct.
I'm trying archive from xcode and Validate App from Organizer is working perfectly (idk if it can be a benchmark, just trying hehe)

i've succeeded before with this build id 67206c1b77aab5fafb1f0249

**Reply by dtrdic** [reply]
hey @ariefrhmn can you please update your yaml by adding build/ios/xcarchive/*.xcarchive to the artifacts section so we can investigate your .xarchive file
```
    artifacts:
      - build/ios/xcarchive/*.xcarchive
```

**Reply by ariefrhmn** [reply]
here is the build Id: 
`6732c75cd1a257bd2b9036f5`

thankss!

**Reply by ariefrhmn** [reply]
oh yaa fyi @dtrdic, when I build using my default scheme/target, the build is successful (build ID: `672eef10d1a257bd2b8627d7`). maybe you can compare with this build?
for now, I can proceed with this method by only changing my environment in the code

**Comment by priitlatt** [comment]
Hi @ariefrhmn. There seems to be quite a bit of noise and sidetracking going on here.

1. Build `672067af8684aa574154f760` in the original report indeed failed because of missing destination specifier `--archive-flags="-destination 'generic/platform=iOS'"`.
2. One destination was specified, then build `672db2d8d1a257bd2b803fd0` failed because of this error:
    ```
    /Users/builder/clone/ios/DiriClinic.xcodeproj: error: Provisioning profile "Diri Care Dev" doesn't match the entitlements file's value for the com.apple.security.application-groups entitlement. Profile qualification is using entitlement definitions that may be out of date. Connect to network to update. (in target 'DiriClinicDevelopment' from project 'DiriClinic')
    ```
2. Now `--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"` was added, which clearly disables signing altogether and we end up with build `672db5f8d1a257bd2b804b8d` and its `[...] DiriClinicDevelopment.app: code object is not signed at all` error, which is very much expected given the build arguments.

Now, to solve the actual issue in point **2.** you'd either have to update your entitlements so that profile's "App Groups" entitlements match with the values that the app requires.

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
